tests/components/smarttub/test_climate.py

```diff
@@ -0,0 +1,83 @@
+"""Test the SmartTub climate platform."""
+from unittest.mock import Mock
+
+import pytest
+
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    HVAC_MODE_HEAT,
+)
+from homeassistant.components.smarttub.climate import (
+    SmartTubThermostat,
+    async_setup_entry,
+)
+from homeassistant.components.smarttub.const import (
+    DEFAULT_MAX_TEMP,
+    DEFAULT_MIN_TEMP,
+    DOMAIN,
+    SMARTTUB_CONTROLLER,
+)
+from homeassistant.components.smarttub.controller import SmartTubController
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from tests.async_mock import create_autospec
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture(name="controller")
+async def mock_controller(hass, coordinator):
```

Comment:
Mock the api, not the controller

Reply:
OK, it sounds like you want these tests to be broader than just unit tests for the module. I can work on that.

Maybe it would be good to add some guidance to https://developers.home-assistant.io/docs/development_testing/ about the scope of tests?

Reply:
As long as the parts that contain mostly logic and no/little device interaction (e.g. `config_flow.py`) have 100% coverage, you don't necessarily need to increase the scope of testing. You just need to make sure that you're mocking the outside, rather than parts of HA itself.